libgraphics/src/utils.cpp

```diff
@@ -292,10 +297,10 @@ bool MCGDashesToSkDashPathEffect(MCGDashesRef self, SkDashPathEffect*& r_path_ef
 	t_dash_effect = NULL;
 	if (t_success)
 	{
-		for (uint32_t i = 0; i < self -> count; i++)
-			t_dashes[i] = MCGFloatToSkScalar(self -> lengths[i]);	
+		for (uint32_t i = 0; i < t_dash_count; i++)
+			t_dashes[i] = MCGFloatToSkScalar(self -> lengths[i % self -> count]);
 	
-		t_dash_effect = new SkDashPathEffect(t_dashes, self -> count, MCGFloatToSkScalar(self -> phase));
+		t_dash_effect = new SkDashPathEffect(t_dashes, (int)t_dash_count, MCGFloatToSkScalar(self -> phase));
```

Comment:
Since the argument you pass to `SkDashPathEffect()` eventually needs to be an `int`, I suggest using `int t_dash_count` with the appropriate overflow checks to avoid an overflowable cast (although if someone's using a dash pattern that long I'm not sure we should worry too much about whether it overflows or not).

````c++
int t_dash_count;
if (t_success)
{
    if (self->count % 2 > 0)
    {
        /* Skia doesn't accept odd-length dash patterns, so repeat any odd-length pattern twice */
        t_dash_count = 2 * self->count;
        t_success = (self->count <= INT_MAX / 2); /* overflow check */
    }
    else
    {
        t_dash_count = self->count;
        t_success = (self->count <= INT_MAX);
    }
}

/* ... */

if (t_success)
{
    for (int i = 0; i < t_dash_count; ++i)
        /* ... etc ... */
}
````